### ISSUE TITLE
v2 aws kms encrypt behavior is different than v1

### ISSUE THREAD

**Issue by Kevmo92** [aws/aws-cli#4994]
With v1 of the awscli I'm able to make a command call like:
```bash
aws kms encrypt --region ${MyRegion} --key-id ${MyKmsKeyId} --plaintext "myEmail@address.com" --output text --query CiphertextBlob
```

However, with v2, running the same command the awscli returns an error of:
`Invalid base64: "myEmail@address.com"`

A current workaround I came across is doing:
```bash
aws kms encrypt --region ${MyRegion} --key-id ${MyKmsKeyId} --plaintext fileb://<(echo "myEmail@address.com") --output text --query CiphertextBlob
```

I'm curious, why is the plaintext invalid unless it's in a file?
It seems that the passed in blob has to meet some new criteria with v2 in order to be valid 🤔 

aws-cli/2.0.0 Python/3.8.1 Darwin/18.7.0 botocore/2.0.0dev4

**Comment by KaibaLopez** [comment]
Hi @Kevmo92 ,
I'll take a look around, but my first impression is that this was the intended behavior, even the documentation example for V1.18 uses a file so not sure if I would count as a bug...
For now I'll mark it as a feature request and check if there was a reason why it no longer works the same way.

**Comment by alFReD-NSH** [comment]
I can replicate this and it looks like a bug to me. Plaintext should not be expected to be base64.

**Comment by KaibaLopez** [comment]
@kevmo92 and @alFReD-NSH ,
As it turns out this was a planned change, refer to the [breaking changes documentation](https://docs.aws.amazon.com/cli/latest/userguide/cliv2-migration.html#cliv2-migration-binaryparam) for more info on the change.

**Comment by shredding** [comment]
tl;dr: add `cli_binary_format=raw-in-base64-out` to your `~/.aws/config` to use plaintext as, well, plaintext.

**Comment by johnboiles** [comment]
Be warned that when using @Kevmo92's super helpful workaround it will add an extra `\n` newline character to the end of your encrypted secret. Don't forget to strip that out in your code.

**Comment by pfandie** [comment]
I stumbled upon the same issue.
but instead of adding `cli_binary_format=raw-in-base64-out` to your config profile(s) like @shredding recommended
you can add the param `--cli-binary-format raw-in-base64-out`
otherwise you won´t be able to decrypt without setting cli-binary-format.

`aws kms encrypt --region ${MyRegion} --key-id ${MyKmsKeyId} --plaintext "string-to-encrypt" --output text --query CiphertextBlob --cli-binary-format raw-in-base64-out`

**Comment by jvepuri** [comment]
How do you decrypt it once it is encrypted on windows

aws kms encrypt  --plaintext "this is a test" --key-id  mykmsid --output text --query CiphertextBlob --cli-binary-format raw-in-base64-out --profile masterorguser > testing.base64

certutil -decode .\testing.base64 testingbase64decoded

Now 

PS C:\courses\awsorganizations\kms> aws kms decrypt --ciphertext-blob fileb:\\testingbase64decoded --key-id arn:aws:kms:us-east-1:347625450273:key/d0cdf03f-1ba1-4b8a-b72b-62c49bf4a46f  --query Plantext --output text  --profile masterorguser | base64 -d

Invalid base64: "fileb:\\testingbase64decoded"

**Comment by jvepuri** [comment]
aws kms encrypt  --plaintext "this is a test" --key-id arn:aws:kms:us-east-1:id --output text --query CiphertextBlob --cli-binary-format raw-in-base64-out --profile masterorguser > testing.base64

PS C:\courses\awsorganizations\kms> certutil -decode .\testing.base64 testingbase64decoded

PS C:\courses\awsorganizations\kms> aws kms decrypt --ciphertext-blob fileb:\\testingbase64decoded --key-id arn:aws:kms:us-east-1:keyid  --cli-binary-format raw-in-base64-out --profile masterorguser

An error occurred (InvalidCiphertextException) when calling the Decrypt operation:


PS C:\courses\awsorganizations\kms> aws kms decrypt --ciphertext-blob fileb:\\testingbase64decoded --key-id arn:aws:kms:us-east-1:keyid  --query Plantext --output text|base64 -d  --profile masterorguser

Invalid base64: "fileb:\\testingbase64decoded"



can you tell how to resolve this?

**Comment by jvepuri** [comment]
aws kms encrypt  --plaintext file:\\masterorgkmssmalltest.txt --key-id arn:aws:kms:us-east-1:keyid --output text --query CiphertextBlob --cli-binary-format raw-in-base64-out --profile masterorguser > base64testing.base64

PS C:\courses\awsorganizations\kms> aws kms decrypt --ciphertext-blob fileb:\\base64testing.base64 --output text --query CiphertextBlob  --profile masterorguser

An error occurred (InvalidCiphertextException) when calling the Decrypt operation:

Can you tell how to resolve this
